scanner/keppel/processor/processor.go

```diff
@@ -0,0 +1,196 @@
+	"github.com/machinebox/graphql"
+	log "github.com/sirupsen/logrus"
+)
+
+type Processor struct {
+	Client *graphql.Client
+}
+
+func NewProcessor(cfg Config) *Processor {
+	return &Processor{
+		Client: graphql.NewClient(cfg.HeurekaUrl),
+	}
+}
+
+func (p *Processor) ProcessRepository(repository models.Repository) (models.Component, error) {
+	var componentRespData struct {
+		Component models.Component `json:"createComponent"`
+	}
+
+	// Create new Component
+	req := graphql.NewRequest(CreateComponentQuery)
+	req.Var("input", map[string]string{
+		"name": repository.Name,
+		"type": "containerImage",
+	})
+
+	err := p.Client.Run(context.Background(), req, &componentRespData)
+
+	return componentRespData.Component, err
+}
+
+func (p *Processor) ProcessManifest(manifest models.Manifest, componentId string) (models.ComponentVersion, error) {
+	var componentVersionRespData struct {
+		ComponentVersion models.ComponentVersion `json:"createComponentVersion"`
+	}
+
+	// Create new ComponentVersion
+	req := graphql.NewRequest(CreateComponentVersionQuery)
+	req.Var("input", map[string]string{
+		"version":     manifest.Digest,
+		"componentId": componentId,
+	})
+
+	err := p.Client.Run(context.Background(), req, &componentVersionRespData)
+
+	return componentVersionRespData.ComponentVersion, err
+}
+
+func (p *Processor) ProcessReport(report models.TrivyReport, componentVersionId string) {
+	for _, result := range report.Results {
+		for _, vulnerability := range result.Vulnerabilities {
+			issue, err := p.GetIssue(vulnerability.VulnerabilityID)
+			if err != nil {
+				log.WithFields(log.Fields{
+					"vulnerabilityID":    vulnerability.VulnerabilityID,
+					"issueID":            issue.ID,
+					"issuePrimaryName":   issue.PrimaryName,
+					"componentVersionID": componentVersionId,
+					"report":             report.ArtifactName,
+				}).WithError(err).Error("Error while getting issue")
+				continue
+			}
+			if issue == nil {
+				log.WithFields(log.Fields{
+					"vulnerabilityID": vulnerability.VulnerabilityID,
+				}).Warning("Issue not found")
+				continue
+				// use this for inserting issues, necessary to test without nvd scanner
+				// i, err := p.CreateIssue(vulnerability.VulnerabilityID, vulnerability.Description)
+				// if err != nil {
+				// 	fmt.Println(err)
+				// 	continue
+				// }
+				// issue = &i
+			}
+			var respData struct {
+				ComponentVersion models.ComponentVersion `json:"addComponentVersionToIssue"`
+			}
+			req := graphql.NewRequest(AddComponentVersionToIssueQuery)
+			req.Var("componentVersionId", componentVersionId)
+			req.Var("issueId", issue.ID)
+			err = p.Client.Run(context.Background(), req, &respData)
+			if err != nil {
+				fmt.Println(err)
+				log.WithFields(log.Fields{
+					"issueID":            issue.ID,
+					"componentVersionID": componentVersionId,
+				}).WithError(err).Error("Could not add component version to issue")
+			}
+		}
+
+	}
+}
```

Comment:
So we do not have a SARIF processor in the first iteration? I though we want to serialize into the Trivy model and move to SARIF directly?!